lib/vector-core/buffers/src/test/common/message.rs

```diff
@@ -14,6 +15,12 @@ impl Message {
     }
 }
 
+impl ByteSizeOf for Message {
+    fn allocated_bytes(&self) -> usize {
+        self.id.size_of()
+    }
```

Comment:
I don't think this is necessary. The `id` does not have allocated data, so it would be included in the base `size_of` implementation. Same for the other implementations AFAICT.